### PR TITLE
Part 3

### PR DESCRIPTION
**Pull request by samyou-softwire**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,5 @@ dist
 .yarn/build-state.yml
 .yarn/install-state.gz
 .pnp.*
+
+.idea
```
Comment (kwunkiuma) [inline .gitignore:131]: 👌

File: src/photos.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+export interface Rover {
+    id:           number;
+    name:         RoverName;
+    landing_date: Date;
+    launch_date:  Date;
+    status:       Status;
+    max_sol:      number;
+    max_date:     Date;
+    total_photos: number;
+    cameras:      CameraElement[];
```
Comment (kwunkiuma) [inline src/photos.ts]: Maybe something like RoverCamera would make more sense? 🙂

File: src/photos.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+export type Status = "active";
+
+
+export function addPhotosEndpoint(router: Router) {
+    router.get('/rovers/:roverName/photos/:camera', async (req, res) => {
+        const { roverName, camera } = req.params;
+        const url = `https://api.nasa.gov/mars-photos/api/v1/rovers/${roverName}/photos`
+
+        const response: { data: PhotosResponse } = await axios.get(url, {
```
Comment (kwunkiuma) [inline src/photos.ts]: Shall we replace this with `axios.get<PhotosResponse>` perhaps? 😄

**Review by kwunkiuma — CHANGES_REQUESTED** [review]
Looks good! Some small nitpicks 😄

File: src/photos.ts (hunk 2 above)
Comment (samyou-softwire) [reply src/photos.ts]: sure 👍,  these names were auto generated

File: src/photos.ts (hunk 3 above)
Comment (samyou-softwire) [reply src/photos.ts]: yep

**Review by kwunkiuma — APPROVED** [review]
Looks good, thanks!

**Pull request merged**